cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -609,7 +609,9 @@ func isAllowedFlag(flagName string) bool {
 		kubeadmcmdoptions.NodeName,
 		kubeadmcmdoptions.KubeconfigDir,
 		kubeadmcmdoptions.UploadCerts,
+		kubeadmcmdoptions.FeatureGatesString,                                       // TODO: remove this line when --feature-gates flag is deprecated and dropped from "kubeadm upgrade apply/plan"
```

Comment:
i think we should disable the flag too. i.e. if the user passes it, it should not do anything.
this aligns with the deprecation in 1.30 to not allow clusterconfig on upgrade.

Reply:
It's no-op now. But we should not break user scripts, so just allow passing this flag with --config in v1.31, and remove it in v1.32

Reply:
agreed